metadata_backend/api/handlers/publish.py

```diff
@@ -516,14 +518,22 @@ async def publish_submission(self, req: Request) -> Response:
         publish_status = {}
         datacite_study = {}
         if "messageBroker" in workflow.endpoints:
-            ingest_msg = {"type": "ingest", "user": external_user_id, "filepath": "filepath"}
-            trigger_ingest = MQPublisher(workflow.get_endpoint_conf("messageBroker", "endpoint"))
-            trigger_ingest.send_message(
-                "ingest",
-                workflow.get_endpoint_conf("messageBroker", "exchange"),
-                ingest_msg,
-                "message_ingestion-trigger",
-            )
+
+            files = await file_operator.read_submission_files(submission_id)
+            for file in files:
+                ingest_msg = {
+                    "type": "ingest",
+                    "user": external_user_id,
+                    "filepath": file["path"],
+                    "encrypted_checksums": file["encrypted_checksums"],
+                }
+                trigger_ingest = MQPublisher(workflow.get_endpoint_conf("messageBroker", "endpoint"))
```

Comment:
not sure, I was thinking the same, but this will be per vhost, and we will have multiple vhosts, probably we can remove the vhost requirement from `MessageBroker` and then require it in `MQPublisher` `send-message` function. That might work.
Should I make the change as part of this ?

Reply:
As you wish